webui/react/src/components/Metadata.tsx

```diff
@@ -0,0 +1,63 @@
+import Button from 'hew/Button';
+import Icon from 'hew/Icon';
+import Message from 'hew/Message';
+import Tree, { TreeDataNode } from 'hew/Tree';
+import { useCallback, useMemo } from 'react';
+
+import { JsonObject, TrialDetails } from 'types';
+import { downloadText } from 'utils/browser';
+import { isJsonObject } from 'utils/data';
+
+import Section from './Section';
+
+export interface Props {
+  trial?: TrialDetails;
+}
+
+export const EMPTY_MESSAGE = 'No metadata found';
+
+const Metadata: React.FC<Props> = ({ trial }: Props) => {
+  const getNodes = useCallback((data: JsonObject): TreeDataNode[] => {
+    return Object.entries(data).map(([key, value]) => {
+      if (isJsonObject(value)) {
+        return { children: getNodes(value), key, title: key };
+      } else if (value) {
+        const stringValue = value.toString();
+        return { children: [{ key: stringValue, title: stringValue }], key, title: key };
+      }
+      return { children: [{ key: 'undefined', title: 'undefined' }], key, title: key };
```

Comment:
this will cause all falsy values to show `'undefined'`, including `false`, `''` and `0`. collapse the missing value branch and use the nullish coalescing operator instead:
```suggestion
      if (isJsonObject(value)) return { children: getNodes(value), key, title: key };
      const stringValue = (value ?? 'undefined').toString();
      return { children: [{ key: stringValue, title: stringValue }], key, title: key };
```